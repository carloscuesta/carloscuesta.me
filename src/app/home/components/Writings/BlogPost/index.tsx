import Image from 'next/image'
import Link from 'next/link'

import { type PostPreview } from 'src/utils/api/blog/mutators'

type Props = {
  post: PostPreview
  isImageLoadedWithPriority: boolean
}

const BlogPost = (props: Props) => (
  <div className="w-full flex-none snap-start px-2 py-2 sm:w-1/2 md:w-1/3">
    <Link
      href={`/blog/${props.post.slug}`}
      title={props.post.title}
      className="h-full rounded-lg outline-none ring-offset-4 focus-visible:ring-2 sm:flex"
    >
      <article>
        <Image
          alt=""
          blurDataURL={props.post.images.preview.lqpi}
          className="mb-3 w-full rounded-lg"
          height={150}
          placeholder="blur"
          priority={props.isImageLoadedWithPriority}
          src={props.post.images.preview.src}
          width={300}
        />

        <div className="grid gap-2">
          <header className="font-semibold">{props.post.title}</header>

          <p className="text-sm opacity-70">{props.post.excerpt}</p>
        </div>
      </article>
    </Link>
  </div>
)

export default BlogPost
