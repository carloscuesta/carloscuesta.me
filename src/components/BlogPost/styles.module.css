.postContainer {
  display: flex;
}

.post {
  border-radius: 2px;
  box-shadow: 0 1px 4px var(--shadowColor);
  color: var(--text);
  margin-bottom: 1em;
  overflow: hidden;
  text-decoration: none;
  transition: all 0.2s ease-out;
}

.post:hover {
  box-shadow: 0 8px 16px var(--shadowColor);
  transform: translateY(-1px);
}

@media (prefers-color-scheme: dark) {
  .post {
    background-color: var(--backgroundSurface);
    box-shadow: none;
  }

  .post:hover {
    background-color: var(--shadowColor);
    box-shadow: none;
  }
}

.postImage {
  filter: blur(5px);
  height: 166px;
  opacity: 0;
  transition: filter 500ms ease-in-out, opacity 500ms ease-in-out;
  width: 100%;
}

.postImage:global(.lazyloaded) {
  filter: blur(0);
  height: auto;
  opacity: 1;
}

.postContent {
  padding: 1.5em;
}

.postPublishedAt {
  color: var(--textSecondary);
  font-size: 0.75em;
  margin-right: 1em;
  text-transform: uppercase;
}

.postExcerpt {
  font-weight: 300;
}
