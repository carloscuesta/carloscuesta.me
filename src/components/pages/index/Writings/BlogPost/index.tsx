import Link from 'next/link'

import { type PostPreview } from 'src/utils/api/blog/mutators'

type Props = {
  post: PostPreview
}

const BlogPost = (props: Props) => (
  <div className="w-full sm:w-1/2 md:w-1/3 flex-none snap-start px-2 py-2">
    <Link
      href={`/blog/${props.post.slug}`}
      title={props.post.title}
      className='sm:flex rounded-lg h-full outline-none focus-visible:ring-2 ring-offset-4'
    >
      <article>
        <img
          className='w-full lazyload rounded-lg mb-3'
          data-src={props.post.images.preview.src}
          src={props.post.images.preview.lqpi}
        />

        <div className='grid gap-2'>
          <header className='font-semibold'>
            {props.post.title}
          </header>

          <p className='opacity-70 text-sm'>
            {props.post.excerpt}
          </p>
        </div>
      </article>
    </Link>
  </div>
  
)

export default BlogPost
