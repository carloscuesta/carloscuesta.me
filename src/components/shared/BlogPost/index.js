// @flow
import React, { type Element } from 'react'
import Link from 'next/link'

import { type PostPreview } from 'src/utils/api/blog/mutators'
import styles from './styles.module.css'

type Props = {
  post: PostPreview
}

const BlogPost = (props: Props): Element<'div'> => (
  <div className={`col-xs-12 col-sm-6 col-md-4 ${styles.postContainer}`}>
    <Link href={`/blog/${props.post.slug}`}>
      <a className={styles.post} title={props.post.title}>
        <article>
          <img
            alt={props.post.title}
            className={`lazyload ${styles.postImage}`}
            data-src={props.post.images.preview.src}
            src={props.post.images.preview.lqpi}
          />

          <div className={styles.postContent}>
            <header>
              <h5 className={styles.postTitle}>{props.post.title}</h5>
            </header>

            <time className={styles.postPublishedAt} dateTime={props.post.datePublished.value}>
              {props.post.datePublished.formatInWords}
            </time>

            <p className={styles.postExcerpt}>
              {props.post.excerpt}
            </p>
          </div>
        </article>
      </a>
    </Link>
  </div>
)

export default BlogPost
