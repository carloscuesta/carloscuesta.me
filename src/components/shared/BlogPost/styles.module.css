.postContainer {
  display: flex;
}

.post {
  border-radius: 2px;
  box-shadow: 0 1px 4px var(--shadowColor);
  color: var(--text);
  margin-bottom: 1em;
  overflow: hidden;
  text-decoration: none;
  transition: all 0.2s ease-out;
}

.post:hover {
  box-shadow: 0 8px 16px var(--shadowColor);
  transform: translateY(-1px);
}

@media (prefers-color-scheme: dark) {
  .post {
    background-color: var(--backgroundSurface);
    box-shadow: none;
  }

  .post:hover {
    background-color: var(--shadowColor);
    box-shadow: none;
  }
}

.postTitle {
  margin: 0;
  margin-bottom: .5em;
}

.postImage {
  filter: blur(5px);
  /* https://stackoverflow.com/questions/27532270/css-blur-weird-shadow-effect */
  transform: translate3d(0, 0, 0);
  transition: filter 200ms ease-in-out;
  width: 100%;
}

.postImage:global(.lazyloaded) {
  filter: blur(0);
}

@media(max-width: 768px) {
  .postImage {
    filter: unset;
    transform: unset;
    transition: unset;
  }

  .postImage:global(.lazyloaded) {
    filter: unset;
  }
}

.postContent {
  padding: 1.5em;
}

.postPublishedAt {
  color: var(--textSecondary);
  font-size: 0.75em;
  margin-right: 1em;
  text-transform: uppercase;
}

.postExcerpt {
  font-weight: 300;
}
