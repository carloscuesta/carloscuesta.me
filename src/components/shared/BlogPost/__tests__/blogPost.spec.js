import renderer from 'react-test-renderer'

import BlogPost from '../index'
import * as stubs from './stubs'

jest.mock('next/link', () => 'a')

describe('BlogPost', () => {
  it('should match BlogPost component', () => {
    const wrapper = renderer.create(<BlogPost {...stubs.props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
