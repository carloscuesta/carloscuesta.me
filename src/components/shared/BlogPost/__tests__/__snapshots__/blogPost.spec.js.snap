// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost should match BlogPost component 1`] = `
<div
  className="col-xs-12 col-sm-6 col-md-4 postContainer"
>
  <a
    as="/blog/post-slug"
    href="/blog/[slug]"
  >
    <a
      className="post"
      title="Post Title"
    >
      <article>
        <img
          alt="Post Title"
          className="lazyload postImage"
          data-src="https://res.cloudinary.com/carloscuesta/image/upload/w_500/v1454754838/kcuq34zbo1yieygmx0y6.png"
          src="https://res.cloudinary.com/carloscuesta/image/upload/t_lqpi-post-preview/v1454754838/kcuq34zbo1yieygmx0y6.png"
        />
        <div
          className="postContent"
        >
          <header>
            <h5
              className="postTitle"
            >
              Post Title
            </h5>
          </header>
          <time
            className="postPublishedAt"
            dateTime="2019-12-15 10:00"
          >
            A year ago
          </time>
          <p
            className="postExcerpt"
          >
            This is the post excerpt that will be shown.
          </p>
        </div>
      </article>
    </a>
  </a>
</div>
`;
